app/forms/household_composition_form.py

```diff
@@ -0,0 +1,112 @@
+        # are considered invalid in python
+        field = StringField(validators=validate_with)
+
+        setattr(NameForm, answer['id'], field)
+
+    return NameForm
+
+
+def serialise_composition_answers(location, data):
+    answers = []
+    for index, person_data in enumerate(data):
+        for answer_id in person_data.keys():
+            answer = Answer(
+                location=location,
+                answer_id=answer_id,
+                answer_instance=index,
+                value=person_data[answer_id],
+            )
+            answers.append(answer)
+    return answers
+
+
+def deserialise_composition_answers(answers):
+    household = {}
+    answer_instance = 0
+
+    while any((a for a in answers if a['answer_instance'] == answer_instance)):
+        instance_answers = (a for a in answers if a['answer_instance'] == answer_instance)
+
+        for instance_answer in instance_answers:
+            composition_id = 'household-{index}-{subfield}'.format(
+                index=answer_instance,
+                subfield=instance_answer['answer_id'],
+            )
+            household[composition_id] = instance_answer['value']
+
+        answer_instance += 1
+    return household
+
+
+def map_field_errors(errors, index):
+    ordered_errors = []
+    for subfield, errorlist in errors.items():
+        answer_id = 'household-{index}-{subfield}'.format(index=index, subfield=subfield)
+        for error in errorlist:
+            ordered_errors.append((answer_id, error))
+    return ordered_errors
+
+
+def generate_household_composition_form(block_json, data, error_messages):
+    class HouseHoldCompositionForm(FlaskForm):
+        household = FieldList(FormField(get_name_form(block_json, error_messages)), min_entries=1)
+
+        def map_errors(self):
+            ordered_errors = []
+
+            if 'household' in self.errors and len(self.errors['household']) > 0:
+                for index, field in enumerate(self.household):
+                    ordered_errors += map_field_errors(field.errors, index)
+            return ordered_errors
+
+        def answer_errors(self, input_id):
+            return [error[1] for error in self.map_errors() if input_id == error[0]]
+
+        def remove_person(self, index_to_remove):
+            popped = []
+
+            while index_to_remove != len(self.household.data):
+                popped.append(self.household.pop_entry())
+
+            popped.reverse()
+
+            for field in popped[1:]:
+                self.household.append_entry(field.data)
+
+        def serialise(self, location):
+            """
+            Returns a list of answers representing the form data
+            :param location: The location to associate the form data with
+            :return:
+            """
+            return serialise_composition_answers(location, self.household.data)
+
+    return HouseHoldCompositionForm(MultiDict(data), meta={'csrf': False})
```

Comment:
ok, raised a card to review CSRF in general: https://trello.com/c/ZppQHbWg/392-review-use-of-csrf-and-whether-it-is-enabled-disabled-etc